crowbar_framework/vendor/assets/javascripts/jquery/dynamicTable.js

```diff
@@ -307,6 +324,44 @@
         optionals: optionals
       });
     });
+
+    self.root.find('tbody select').live('change', function(event) {
+      self.prepareJson();
+
+      var data = self.json;
+      var namespace = $(this).data('update').toString().split('/');
+
+      while (namespace.length > 1) {
+        var current_namespace = namespace.shift();
+
+        if (!data[current_namespace]) {
+          data[current_namespace] = {};
+        }
+
+        data = data[current_namespace];
+      }
+
+      if ($(this).data('type') == "array-string") {
+        // for multi-selects
+        var value = $(this).val();
+      } else {
+        var value = self.parseValue(
+          $(this).data('type'),
+          $(this).val()
+        );
+      }
+
+      data[namespace.shift()] = value;
```

Comment:
not true, the declaration of javascript variables are moved to the execution context:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var

but I wonder if we can do that more clearly doing the `var values = '';` outside the `if`